database1/pgQueryTests.js

```diff
@@ -0,0 +1,64 @@
+const pg = require('./pgQueryTests.js')
+const { Client } = require('pg');
+
+const client = new Client({
+  user: 'clairemelbourne',
+  database: 'similar_prod'
+});
+
+client.connect();
+
+//get request from 14000000
+const getOne = (productId, res) => {
+  const priceText = `SELECT price FROM products WHERE productid = ${productId}`;
+  client.query(priceText)
+  .then((result) => {
+    const price = result.rows[0].price
+    const productsText = `SELECT * FROM products WHERE ${getRange(productId, null)} AND price <= ${price + 50} AND price <= ${price - 50} LIMIT 8`;
+    client.query(productsText)
+    .then((result) => {
+      console.log("results of query", result.rows);
+    })
+  })
+  .catch((err) => {
+    console.error(err.stack);
+    client.end();
+  })
+}
+
+const create = (req, res) => {
+  console.log("request body", req.body);
+  const b = req.body
+  const idText = `SELECT MAX(productid) FROM products WHERE ${getRange(null, b.category)}`;
+  client.query(idText)
+  .then((result) => {
+  console.log("id", result.rows[0].max)
+  var productId = result.rows[0].max + 1;
+    var text = `INSERT INTO products (productid, name, category, price, rating, imageurl, onsale) VALUES ($1, $2, $3, $4, $5, $6, $7) RETURNING *`;
+    var values = [productId, b.name, b.category, b.price, b.rating, b.imageUrl, b.onSale];
+    client.query(text, values)
+    .then((result) => {
+      const message = `${result.rows[0].name} added to products`
+      res.status(201).send(message);
+    })
+    .catch(err => {
```

Comment:
I've never used more than one catch block. Does JS know which block it should go to based on where it is in the execution of this promise chain? I would imagine you could also accomplish this with conditionals within a single catch block.

Reply:
ah good question -- sorry I didn't actually mean for this file to be in the pr, it is very rough! but to answer your question, yes, in this instance it did seem to be working as a nested .then() .catch(). I added it because it wasn't being caught by my outter catch for some reason but I agree its probably not best practice!